sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/TablesJacksonSerializer.java

```diff
@@ -140,9 +146,13 @@ public void serialize(Object object, SerializerEncoding encoding, OutputStream o
             values.add(singleValue);
         }
 
-        return (U) new TableEntityQueryResponse()
-            .setOdataMetadata(odataMetadata)
-            .setValue(values);
+        if (type == TableEntityQueryResponse.class) {
+            return (U) new TableEntityQueryResponse()
+                .setOdataMetadata(odataMetadata)
+                .setValue(values);
+        } else {
+            return (U) singleValue;
```

Comment:
Suspicious about this. If `type` is a Map, does that mean that we are always in the single value case? If so, I guess that's okay but I would want to see that case handled separately up at L140, e.g.

```java
if (type instanceof ParameterizedType && ((ParameterizedType) type).getRawType() == Map.class) {
    // We're always in the single-value case if the type to deserialize is a Map
    return (U) singleValue;
} else if (singleValue != null) {
   ...
```

Reply:
Modified this with Brandon's help.